### ISSUE TITLE
Make client CDI plugable

### ISSUE THREAD

**Issue by phillip-kruger** [smallrye/smallrye-graphql#153]
Make the client CDI plugable (like the server)

**Comment by t1** [comment]
You mean for the Client?

**Comment by phillip-kruger** [comment]
Yes, sorry did not add a description yet, just wanted to remind myself. But the DI should be (if possible) pluggable.
I have made a change to the server code where I removed CDI from the impl, and delegate the loading of instances to SPI. 
I added a CDI SPI module (that will be the one we use mostly) but (at least theoretically) we should be able to also use Guice or Spring DI.

There are some other dependencies we possible should also move to be loaded via SPI, like json-b and json-p.

I moved the config out too, so for example, Servlet implementation now provide the config. That should make this lib usable in other frameworks.
